### ISSUE TITLE
Support generic functions in `MethodBuilder`

### ISSUE THREAD

**Issue by zoechi** [dart-lang/code_builder#95]
```dart
T myFunc<T>(T x) {... }
```
Passing generic parameters to method/function calls was added recently, but I couldn't find anything to add them to `MethodBuilder`.

**Comment by matanlurey** [comment]
I'd like to do a re-design of generics (instead of relying on `generics: [ ... ]`, i.e.

```dart
abstract class SupportsGenericTypes {
  void addGenericType(...);
}
```

But I'm afraid this won't happen by 1.0, as I'd like to release a stable version.

**Comment by matanlurey** [comment]
Implemented on the 2.0 branch:

https://github.com/dart-lang/code_builder/tree/v2
